charms/jupyter-controller/tests/test_operator.py

```diff
@@ -11,7 +11,7 @@ def harness():
 
 def test_not_leader(harness):
     harness.begin()
-    assert harness.charm.model.unit.status == WaitingStatus("Waiting for leadership")
+    assert harness.charm.model.unit.status == MaintenanceStatus("")
```

Comment:
Given that this status check is now done in `main` you will have to call `harness.begin_with_inital_hooks` else you never reach that logic.

Reply:
Updated.